modules/op-deploying-api-service-for-tekton-hub.adoc

```diff
@@ -0,0 +1,186 @@
+
+[discrete]
+.Prerequisites
+* Ensure that you are in the `<project_root>/config/` directory, where `<project_root>` is the top level directory of the cloned {tekton-hub} repository.
+
+[discrete]
+.Procedure
+. Set up route or ingress for the API.
++
+[source,terminal]
+----
+$ oc apply -f 04-openshift/40-api-route.yaml -f 04-openshift/40-auth-route.yaml -n tekton-hub
+----
+
+. Create an OAuth application with your Git repository hosting service provider.
+* For a GitHub OAuth application, set the `Homepage URL` and the `Authorization callback URL` as `<auth-route>`.
+* For a GitLab OAuth application, set the `REDIRECT_URI` as `<auth-route>/auth/gitlab/callback`.
+* For a BitBucket OAuth application, set the `Callback URL` as `<auth-route>`.
+
+. Edit the `02-api/20-api-secret.yaml` file to update the API secret.
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: tekton-hub-api
+  namespace: openshift-pipelines <1>
+type: Opaque
+stringData:
+  GH_CLIENT_ID: '' <2>
+  GH_CLIENT_SECRET: '' <3>
+  GHE_URL: '' <4>
+  GL_CLIENT_ID: '' <5>
+  GL_CLIENT_SECRET: '' <6>
+  GLE_URL: '' <7>
+  BB_CLIENT_ID: '' <8>
+  BB_CLIENT_SECRET: '' <9>
+  JWT_SIGNING_KEY: '' <10>
+  ACCESS_JWT_EXPIRES_IN: '' <11>
+  REFRESH_JWT_EXPIRES_IN: '' <12>
+  AUTH_BASE_URL: '' <13>
+----
+<1> The target namespace mentioned in the `TektonHub` CR; default is `openshift-pipelines`.
+<2> The Client ID from the GitHub OAuth application.
+<3> The Client Secret from the GitHub OAuth application.
+<4> GitHub Enterprise URL, if you are authenticating using GitHub Enterprise. Do not provide the URL to the catalog as a value for this field.
+<5> The Client ID from the GitLab OAuth application.
+<6> The Client Secret from the GitLab OAuth application.
+<7> GitLab Enterprise URL, if you are authenticating using GitLab Enterprise. Do not provide the URL to the catalog as a value for this field.
+<8> The Client ID from the BitBucket OAuth application.
+<9> The Client Secret from the BitBucket OAuth application.
+<10> A long, random string used to sign the JSON Web Token (JWT) created for users.
+<11> Add the time limit after which the access token expires. For example, `1m`, where `m` denotes minutes. The supported units of time are seconds (`s`), minutes (`m`), hours (`h`), days (`d`), and weeks (`w`).
+<12> Add the time limit after which the refresh token expires. For example, `1m`, where `m` denotes minutes. The supported units of time are seconds (`s`), minutes (`m`), hours (`h`), days (`d`), and weeks (`w`). Ensure that the expiry time set for token refresh is greater than the expiry time set for token access.
+<13> Route URL for the OAuth application.
++
+[NOTE]
+====
+Use the fields related to Client ID and Client Secret for any one of the supported Git repository hosting service providers.
+====
+
+. Update the API config map.
+.. Update the `config.yaml` file to include at least one user with the following permissions:
```

Comment:
We should explain why admin should give us their Git provider credentials

Reply:
@PuneetPunamiya - Can you please comment on this? Although this file is now unavailable, we can use the clarification in other places.

Reply:
The main reason user should give his/her git credentials in config.yaml is because with the `catalog-refresh` scopes user can load all the resources from the catalog in the database